library/ix-dev/community/terraria/questions.yaml

```diff
@@ -0,0 +1,316 @@
+            enum:
+              - value: image
+                description: TShock
+              - value: vanillaImage
+                description: Vanilla
+        - variable: worldName
+          label: World Name
+          description: |
+            Sets the world's name.</br>
+            Only applies to newly generated worlds.
+          schema:
+            type: string
+            default: My-World
+            required: true
+            immutable: true
+        - variable: worldSeed
+          label: World Seed
+          description: |
+            Sets the world's seed.</br>
+            Only applies to newly generated worlds.
+          schema:
+            type: string
+            default: ''
+            immutable: true
+        - variable: worldEvil
+          label: World's Evil
+          description: |
+            Sets the world's evil state.</br>
+            Only applies to newly generated worlds.
+          schema:
+            type: string
+            default: random
+            required: true
+            immutable: true
+            enum:
+              - value: random
+                description: Random
+              - value: corrupt
+                description: Corrupt
+              - value: crimson
+                description: Crimson
+        - variable: worldSize
+          label: World Size
+          description: |
+            Sets the world's size.</br>
+            Only applies to newly generated worlds.
+          schema:
+            type: string
+            default: small
+            required: true
+            immutable: true
+            enum:
+              - value: small
+                description: Small (1)
+              - value: medium
+                description: Medium (2)
+              - value: large
+                description: Large (3)
+        - variable: worldDifficulty
+          label: World Difficulty
+          description: |
+            Sets the world's difficulty.</br>
+            Only applies to newly generated worlds.
+          schema:
+            type: string
+            default: normal
+            required: true
+            immutable: true
+            enum:
+              - value: normal
+                description: Normal (0)
+              - value: expert
+                description: Expert (1)
+              - value: master
+                description: Master (2)
+              - value: journey
+                description: Journey (3)
+        - variable: maxPlayers
+          label: Maximum Players
+          description: Starts the server with the given player count as the maximum.
+          schema:
+            type: int
+            default: 8
+            min: 1
+            max: 255
+            required: true
+        - variable: password
+          label: Server Password
+          description: Sets the server password.
+          schema:
+            type: string
+            private: true
+            default: ''
```

Comment:
If this is required please have it as such and `empty=False`

Reply:
It's optional!